js/src/quantity-selector.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * --------------------------------------------------------------------------
+ * Boosted (v5.1.3): quantity-selector.js
+ * Licensed under MIT (https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/LICENSE)
+ * --------------------------------------------------------------------------
+ */
+
+import { defineJQueryPlugin } from './util/index'
+import EventHandler from './dom/event-handler'
+import BaseComponent from './base-component'
+import SelectorEngine from './dom/selector-engine'
+
+/**
+ * Constants
+ */
+
+const NAME = 'quantityselector'
+const DATA_KEY = 'bs.quantityselector'
+const EVENT_KEY = `.${DATA_KEY}`
+const DATA_API_KEY = '.data-api'
+
+const EVENT_LOAD_DATA_API = `load${EVENT_KEY}${DATA_API_KEY}`
+const EVENT_CHANGE_DATA_API = `change${EVENT_KEY}${DATA_API_KEY}`
+const EVENT_CLICK_DATA_API = `click${EVENT_KEY}${DATA_API_KEY}`
+
+const SELECTOR_STEP_UP_BUTTON = '[data-bs-step="up"]'
+const SELECTOR_STEP_DOWN_BUTTON = '[data-bs-step="down"]'
+const SELECTOR_COUNTER_INPUT = '[data-bs-step="counter"]'
+const SELECTOR_INPUT_GROUP = '.input-group'
+
+/**
+ * Class definition
+ */
+
+class QuantitySelector extends BaseComponent {
+  // Getters
+  static get NAME() {
+    return NAME
+  }
+
+  // Public
+  ValueOnLoad(element) {
+    const counterInput = element.querySelector(SELECTOR_COUNTER_INPUT)
+    const btnUp = element.querySelector(SELECTOR_STEP_UP_BUTTON)
+    const btnDown = element.querySelector(SELECTOR_STEP_DOWN_BUTTON)
+
+    const min = counterInput.getAttribute('min')
+    const max = counterInput.getAttribute('max')
+    const step = Number(counterInput.getAttribute('step'))
```

Comment:
Could some variables here become attributes ?

Reply:
Not really because of 'counterInput' which is defined inside the function and not always the same. Don't hesitate if you have an idea about how to refactor and optimize the component without affecting its performance.